components/dashboard/src/settings/Integrations.tsx

```diff
@@ -199,77 +179,435 @@ function GitProviders() {
+            case "read:org": return "Read-only access to organizations (used to suggest organizations when forking a repository)";
+            case "workflow": return "Allow updating GitHub Actions workflow files";
+            // GitLab
+            case "read_user": return "Read-only access to your email addresses";
+            case "api": return "Allow making API calls (used to set up a webhook when enabling prebuilds for a repository)";
+            case "read_repository": return "Read/write access to your repositories";
+            // Bitbucket
+            case "account": return "Read-only access to your account information";
+            case "repository": return "Read-only access to your repositories (note: Bitbucket doesn't support revoking scopes)";
+            case "repository:write": return "Read/write access to your repositories (note: Bitbucket doesn't support revoking scopes)";
+            case "pullrequest": return "Read access to pull requests and ability to collaborate via comments, tasks, and approvals (note: Bitbucket doesn't support revoking scopes)";
+            case "pullrequest:write": return "Allow creating, merging and declining pull requests (note: Bitbucket doesn't support revoking scopes)";
+            case "webhook": return "Allow installing webhooks (used when enabling prebuilds for a repository, note: Bitbucket doesn't support revoking scopes)";
+            default: return "";
+        }
+    }
+
     return (<div>
-        <Modal visible={!!diconnectModal} onClose={() => setDisconnectModal(undefined)}>
-            <h3>You are about to disconnect {diconnectModal?.provider.host}</h3>
-            <div>
-                <button onClick={() => disconnect(diconnectModal?.provider!)}>Proceed</button>
-            </div>
-        </Modal>
 
-        <Modal visible={!!editModal} onClose={() => setEditModal(undefined)}>
-            <div>
-                {editModal && (<React.Fragment>
-                    <h3>Permissions granted</h3>
-                    <div>
-                        {editModal && editModal.provider.scopes!.map(scope => (
-                            <div key={`scope-${scope}`}>
-                                <CheckBox
-                                    name={scope}
-                                    desc={scope}
-                                    title={scope}
-                                    key={`scope-checkbox-${scope}`}
-                                    checked={editModal.nextScopes.has(scope)}
-                                    disabled={editModal.provider.requirements?.default.includes(scope)}
-                                    onChange={onChangeScopeHandler}
-                                ></CheckBox>
-                            </div>
-                        ))}
+        {diconnectModal && (
+            <Modal visible={true} onClose={() => setDisconnectModal(undefined)}>
+                <h3 className="pb-2">Disconnect Provider</h3>
```

Comment:
**issue**: When you are trying to disconnect the only remaining provider we should show an error but it should be fine to get this is as is and add this in a later iteration.

Reply:
☝🏻 that's the mentioned confusion. let's simplify by not rendering the menu item if that would be an error case anyways.

Reply:
Aha! We could also disable the item when this is the only provider left but hiding also works. Let's do this!